dpctl/_sycl_device.pyx

```diff
@@ -2045,6 +2042,63 @@ cdef class SyclDevice(_SyclDevice):
                 else:
                     return str(relId)
 
+    def get_unpartitioned_parent_device(self):
+        """ get_unpartitioned_parent_device(self)
+
+        Returns the unpartitioned parent device of this device, or None for a
+        root device.
+
+        Returns:
+            dpctl.SyclDevice:
+                A parent, unpartitioned :class:`dpctl.SyclDevice` instance if
+                the device is a sub-device, ``None`` otherwise.
+        """
+        cdef DPCTLSyclDeviceRef pDRef = NULL
+        cdef DPCTLSyclDeviceRef tDRef = NULL
+        pDRef = DPCTLDevice_GetParentDevice(self._device_ref)
+        if pDRef is NULL:
+            return None
```

Comment:
In this case it would make sense to return `self`. This way the returned value is always useful.